modules/reindex/src/test/java/org/elasticsearch/index/reindex/DeleteByQueryConcurrentTests.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.termQuery;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.equalTo;
+
+public class DeleteByQueryConcurrentTests extends ReindexTestCase {
+
+    public void testConcurrentDeleteByQueriesOnDifferentDocs() throws Throwable {
+        createIndex("test");
+        ensureGreen();
+
+        final Thread[] threads =  new Thread[scaledRandomIntBetween(2, 5)];
+        final long docs = randomIntBetween(1, 50);
+        for (int i = 0; i < docs; i++) {
+            for (int j = 0; j < threads.length; j++) {
+                index("test", "test", String.valueOf(i * 10 + j), "field", j);
+            }
+        }
+        refresh();
+        assertHitCount(client().prepareSearch("test").setSize(0).get(), docs * threads.length);
+
+        final CountDownLatch start = new CountDownLatch(1);
+        final AtomicReference<Throwable> exceptionHolder = new AtomicReference<>();
```

Comment:
So it turns out that RandomizedRunner sets an uncaught exception handler which should make this not required. I think.